diff --git a/modules/content_multigroup/content_multigroup.node_view.inc b/modules/content_multigroup/content_multigroup.node_view.inc
index 5e3627c..ca9ae14 100644
--- a/modules/content_multigroup/content_multigroup.node_view.inc
+++ b/modules/content_multigroup/content_multigroup.node_view.inc
@@ -73,7 +73,7 @@ function _content_multigroup_fieldgroup_view(&$node, &$element, $group, $context
     foreach (array_keys($group_fields) as $field_name) {
       if (isset($node->content[$field_name])) {
         $node_copy->content[$field_name]['field']['items'] = array(
-          $delta => isset($node->content[$field_name]['field']['items'][$delta]) ? $node->content[$field_name]['field']['items'][$delta] : NULL,
+          0 => isset($node->content[$field_name]['field']['items'][$delta]) ? $node->content[$field_name]['field']['items'][$delta] : NULL,
         );
         $element[$delta][$field_name] = $node_copy->content[$field_name];
         $element[$delta][$field_name]['#delta'] = $delta;
