diff --git a/icl_translate/icl_translate.module b/icl_translate/icl_translate.module
index b7f650c..5c85357 100644
--- a/icl_translate/icl_translate.module
+++ b/icl_translate/icl_translate.module
@@ -30,7 +30,7 @@ function icl_translate_init() {
         'permission' => ICL_TRANSLATE_DO_JOBS,
     )));
   }
-  if (variable_get('icl_translate_manager_role', 0) == 0) {
+  if (variable_get('icl_manager_role', 0) == 0) {
     _icl_wrapper_create_roles(array('icanlocalize manager' => array(
         'title' => 'ICanLocalize manager',
         'variable' => 'icl_manager_role',
