diff --git a/nd_cck/nd_cck.module b/nd_cck/nd_cck.module
index e1906a8..7e2c6cc 100644
--- a/nd_cck/nd_cck.module
+++ b/nd_cck/nd_cck.module
@@ -272,7 +272,7 @@ function theme_nd_cck_node_formatter($element) {
         _nodereference_titles($element['#item']['nid'], $referenced_node->title);
         $page = FALSE;
         $teaser = TRUE;
-        if ($build_mode == 'full') {
+        if (TRUE/*$build_mode == 'full'*/) {
           $page = TRUE;
           $teaser = FALSE;
         }
