Index: smtp.module
===================================================================
RCS file: /cvs/drupal-contrib/contributions/modules/smtp/smtp.module,v
retrieving revision 1.17.2.23
diff -u -p -r1.17.2.23 smtp.module
--- smtp.module	21 Dec 2010 16:15:26 -0000	1.17.2.23
+++ smtp.module	1 Jan 2011 15:14:55 -0000
@@ -384,9 +384,9 @@ function smtp_drupal_mail_wrapper($messa
 
   if ($from == NULL || $from == '') {
     // If from e-mail address is blank, use smtp_from config option.
-    if ($from = variable_get('smtp_from', '') == '') {
+    if (($from = variable_get('smtp_from', '')) == '') {
       // If smtp_from config option is blank, use site_email.
-      if ($from = variable_get('site_email', '') == '') {
+      if (($from = variable_get('site_email', '')) == '') {
         drupal_set_message(t('There is no submitted from address.'), 'error');
         watchdog('smtp', 'There is no submitted from address.', array(), WATCHDOG_ERROR);
         return FALSE;
